PDF Rendering Quality (#1426)

tc-wleite opened the pull request:
As discussed in #1426, if it is not possible to upgrade IcePDF because of its dependencies, maybe we could apply only the second commit (https://github.com/sepinf-inc/IPED/commit/c9244853c8196fc58f44349e0e0ad5b16c644493), which should help when using a UI scale != 1.0.

lfcnassif commented:
I think it is, thank you @tc-wleite!

lfcnassif commented:
@tc-wleite I removed jpeg2000 because of its license, it can be still include into plugins folder by user. Furthermore, I tested its 1.4.0 version again with a very small set of jpeg2000 files I have and the results are still worse than imagemagick. This is a difficult choice because imagemagick is better, but it doesn't work for embedded jpeg2000 into PDFs as it is not a imageio plugin.

I also removed bcprov-ext-jdk15on because it was duplicating a lot of classes already included in bcprov-jdk15on (plus some uncommon algorithm implementation).

Finally, I tried to upgrade imageio-tiff to 3.9.3 because I saw **a lot** of fixes in TwelveMonkeys release notes regarding TIF decoding, but it didn't work, possibly because I kept TwelveMonkeys base libraries in 3.8.x to avoid possibly breaking psd, webp, etc. So I just upgraded it to 3.8.3 version which already has some TIF related fixes.

lfcnassif commented:
> So I just upgraded it to 3.8.3 version which already has some TIF related fixes.

As you tested 3.8.2 with a large set of TIFs (I tested with a small set) and this is a minor upgrade, I'm assuming it is ok.

tc-wleite commented:
> @tc-wleite I removed jpeg2000 because of its license, it can be still included into plugins folder by user. Furthermore, I tested its 1.4.0 version again with a very small set of jpeg2000 files I have and the results are still worse than imagemagick. This is a difficult choice because imagemagick is better, but it doesn't work for embedded jpeg2000 into PDFs as it is not a imageio plugin.
> 
> I also removed bcprov-ext-jdk15on because it was duplicating a lot of classes already included in bcprov-jdk15on (plus some uncommon algorithm implementation).
> 
> Finally, I tried to upgrade imageio-tiff to 3.9.3 because I saw **a lot** of fixes in TwelveMonkeys release notes regarding TIF decoding, but it didn't work, possibly because I kept TwelveMonkeys base libraries as 3.8.x to avoid possibly breaking psd, webp, etc. So I just upgraded it to 3.8.3 version which already has some TIF related fixes.

Handing all these dependencies is not simple at all. 
I agree with all the changes you made.
